doc/reference_doc_guidelines.md

```diff
@@ -0,0 +1,295 @@
+# Azure CLI reference and conceptual content guidelines
+
+## General guidelines
+
+General guidelines apply to both reference and conceptual content.
+
+- Develop and test all code blocks in both a Bash and PowerShell environment.  When commands have syntax differences between environments, create one example for Bash and another for PowerShell.  For copyability, use different code blocks with a notation.  Here is an example:
+
+    ```azurecli-interactive
+    az command –-p1 written-for-Bash 
+    ```
+
+    ```azurecli-interactive
+    # This same command can be run in PowerShell, but with a slight syntax difference 
+    az command –-p1 written-for-PowerShell-showing-quoting-differences 
+    ```
+
+  For more information on syntax differences between Bash and PowerShell, see these articles:
+    - [Quoting issues with PowerShell](https://github.com/Azure/azure-cli/blob/dev/doc/quoting-issues-with-powershell.md)
+    - [Use quotation marks in Azure CLI parameters](./use-cli-effectively.md#use-quotation-marks-in-parameters)
+    - Compare syntax of CMD, PowerShell and Bash in [Query command output using JMESPath](./query-azure-cli.md)
+    - [Error handling for the Azure CLI in PowerShell](./use-cli-effectively.md#error-handling-for-azure-cli-in-powershell)
+
+    **Do not replicate code blocks for line break differences.**  The new Azure CLI code block <kbd>Copy</kbd> button removes Bash and PowerShell line break characters. Only replicate code blocks for syntax differences.
+
+- Provide **\```output** code blocks. When developing scripts, customers will likely combine two or more create, read, update and delete operations. With no documentation about the format of the returned object, customers have no choice but to run each command one by one and analyze the response.
```

Comment:
Providing documentation for output JSON can be challenging. We previously had this conversation in https://github.com/Azure/azure-cli/issues/17332.

Reply:
@jiasli, understood, but this continues to be a request by @dcaro.  Let's compromise and I'll move this to the conceptual requirements section until there is both a **plan** and a **place** for this in the source code.